### PR TITLE
Adapt wazuhapp and Wazuh dashboard to install the Wazuh CheckUpdates and Core plugins

### PR DESCRIPTION
**Pull request by rauldpm**
|Related issue|
|---|
|closes https://github.com/wazuh/wazuh-packages/issues/2589|

## Description

This pull request adapts the APP and Wazuh dashboard packages creation according to the new Wazuh plugins

## Logs example

See https://github.com/wazuh/wazuh-packages/issues/2589#issuecomment-1811347427

## Tests

- Build the package in any supported platform
  - [x] Linux
- [x] Package installation
- [x] Package upgrade
- [x] Package remove

- Tests for Linux RPM
  - [x] Build the package for x86_64
- Tests for Linux deb
  - [x] Build the package for x86_64

### CODE REVIEW

File: wazuhapp/opensearch-dashboards/Docker/Dockerfile [hunk 1]
```diff
@@ -8,7 +8,10 @@ RUN rpm --import /etc/pki/rpm-gpg/RPM-GPG-KEY-centostesting && \
     curl -sL https://rpm.nodesource.com/setup_10.x | bash - && \
     yum install nodejs git gcc gcc-c++ make sudo zip python3 -y && \
     alternatives --set python /usr/bin/python3 && \
-    npm install -g n 
+    npm install -g n
+
+RUN yum install -y \
```
Comment (jnasselle) [inline wazuhapp/opensearch-dashboards/Docker/Dockerfile:13]: Any reason to keep this in a separate RUN command?

**Review by jnasselle — APPROVED** [review]
LGTM!

File: stack/dashboard/deb/build_package.sh [hunk 2]
```diff
@@ -111,19 +126,47 @@ build() {
...
-    echo "    --dont-build-docker        [Optional] Locally built docker image will be used instead of generating a new one."
-    echo "    --future                   [Optional] Build test future package 99.99.0 Used for development purposes."
-    echo "    -h, --help                 Show this help."
-    echo
+    echo -e ""
+    echo -e "NAME"
+    echo -e "        $(basename "$0") - Build Wazuh dashboard base file."
+    echo -e ""
+    echo -e "SYNOPSIS"
+    echo -e "        $(basename "$0") -a | -m | -u | -c | -s | -b | -f | -r | -h"
```
[the remark this answers is not in the document]
Comment (rauldpm) [reply stack/dashboard/deb/build_package.sh:134]: This is the expected output as it was decided a long time ago, the same applies to the OVA and other packages that have been updated before this change, this was a decision from the management team.

https://github.com/wazuh/wazuh-packages/blob/master/ova/generate_ova.sh#L33-L61

File: wazuhapp/opensearch-dashboards/generate_wazuh_app.sh [hunk 3]
```diff
@@ -21,16 +21,29 @@ OUTDIR="${CURRENT_PATH}/output/"
...
+    echo -e "                [Required] Select Git branch or tag."
+    echo -e ""
+    echo -e "        -s, --store <path>"
+    echo -e "                [Optional] Set the destination path of package, by default /tmp/wazuh-app."
+    echo -e ""
+    echo -e "        -r, --revision <rev>"
+    echo -e "                [Optional] Package revision."
+    echo -e ""
+    echo -e "        -c, --checksum <path>"
+    echo -e "                [Optional] Generate checksum."
```
[the remark this answers is not in the document]
Comment (rauldpm) [reply wazuhapp/opensearch-dashboards/generate_wazuh_app.sh:42]: Same, the default value was removed unless it was necessary to the user (as the output directory)

File: wazuhapp/opensearch-dashboards/Docker/Dockerfile (hunk 1 above)
Comment (rauldpm) [reply wazuhapp/opensearch-dashboards/Docker/Dockerfile:13]: Actually, no, I didn't see the other `yum install`, but this dependency is not necessary to build the package itself but to configure the packages names, I think it should be separately

**Review by davidjiglesias — APPROVED** [review]

**Pull request merged**